Neos.Flow/Classes/Mvc/Routing/TestingRoutesProvider.php

```diff
@@ -0,0 +1,57 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Mvc\Routing;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Extends the routing to be able to add custom routes at runtime.
+ *
+ * @internal implementation detail. Please use {@see FunctionalTestCase::registerRoute} instead.
+ */
+#[Flow\Scope("singleton")]
+final class TestingRoutesProvider implements RoutesProviderInterface
```

Comment:
as discussed with @mficzel this class resides here correctly.

`Neos\Flow\Tests` has no proxy building and the `TestingPrivilegeManager` also lives in `Classes`.